AutoPartsShop.WebUI/Pages/ProductDetails.razor

```diff
@@ -0,0 +1,54 @@
+﻿@page "/ProductDetails/{Id:int}"
+@using AutoPartsShop.Models.DTOs
+@inherits ProductDetailsBase
+
+@if (Product == null && ErrorMessage == null)
+{
+    <DisplaySpinner />
+}
+else if (ErrorMessage != null)
+{
+    <DisplayError ErrorMessage="@ErrorMessage"></DisplayError>
+}
+else
+{
+    <h3 class="mb-5">Product Details</h3>
+    <div class="row">
+        <div class="col-md-6 mb-4">
+            <img class="preview-img" src="@Product.ImageURL" alt="@Product.ProductName">
+        </div>
+        <div class="col-md-6">
+            <h3>@Product.ProductName</h3>
+            <p class="mb-4">@Product.Description</p>
+            <p class="mb-4">
+                <b>
+                    @Product.Price.ToString("C")&nbsp;(@Product.Qty items in stock)
+                </b>
+            </p>
+            <div>
+                <button class="btn btn-success"
+                @onclick="() => (AddToCart_Click( new CartItemToAddDTO
+                {
+                    CartId = HardCoded.CartId,
+                    ProductId = Product.Id,
+                    Qty = 1
+                }))">
+                    <b>Add to Cart</b>
+                </button>
+            </div>
+        </div>
+    </div>
+}
+
+@code {
+    private async Task AddToCart()
```

Comment:
This method seems to be unused. Above you are using the `AddToCart_Click` one so either you remove this one or the other one.